### DISCUSSION TITLE
Extract values of variables that depends on uncertain variables

### DISCUSSION THREAD

**Discussion by GalPerelman** [yalmip/YALMIP#1261] [General discussion] [answered]
Is it possible to obtain the values of variables that depend on uncertainty variables?
For example, in adjustable robust optimization, the decision variables are affine functions of the uncertain parameters
A common affine modeling is linear decision rules (LDR)
In this case, I'm looking for the LDR coefficients. By multiplying the coefficients with a sample of the uncertain parameter I will get the decision variables values

An example code:

```
tau = sdpvar;
x = sdpvar(T,1);
d = sdpvar(T,1);
constr = [(1-theta)*d0 <= d <= (1+theta)*d0, uncertain(d)];

for t=2:T
    yalmip('setdependence',x(t), d(1:t-1));
end

constr = [constr, x >= 0];
constr = [constr, x <= 1800];
for t=1:T
    constr = [constr, v + sum(x(1:t)) - sum(d(1:t)) >= vmin];
    constr = [constr, v + sum(x(1:t)) - sum(d(1:t)) <= vmax];
end
constr = [constr, sum(x(1:T)) - sum(d(1:T)) >= 0];
constr = [constr, c * x <= tau];

objective = tau;

ops = sdpsettings('robust.auxred','affine', 'verbose', 0, 'debug', 1);
diagnostics = solvesdp(constr, objective, ops);
value(objective)
xx = value(x);
```

```xx``` gets Nan values because of the dependency on d

**Comment by johanlofberg** [comment, marked as answer]
No, you would have to model it manually https://yalmip.github.io/example/robustmpc/ (recommended anyway since that internal support is something I don't even remember I added, hence very basic and experimental)
